File: 125. Valid Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s = s.lower()
        l = 0
        r = len(s)-1
        while l <= r:
            if s[l].isalnum() and s[r].isalnum():
                if s[l] == s[r]:
                    l += 1
                    r -= 1
                else:
                    return False
            elif s[l].isalnum() and not s[r].isalnum():
                r -= 1
            elif not s[l].isalnum() and s[r].isalnum():
                l += 1
            else:
                l += 1
                r -= 1
        return True

File: 125. Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int l = 0, r = s.length()-1;
        locale loc;
        for (int i=0; i < s.length(); i++)
            s[i] = tolower(s[i], loc);
        while (l <= r)
        {
            if (isalnum(s[l]) && isalnum(s[r]))
            {
                if (s[l] == s[r])
                {
                    l ++;
                    r --;
                }
                else
                    return false;
            }
            else if (!isalnum(s[l]) && isalnum(s[r]))
                l ++;
            else if (isalnum(s[l]) && !isalnum(s[r]))
                r --;
            else
            {
                l ++;
                r --;
            }
        }
        return true;
    }
};
